src/core/gui/SearchBar.cpp

```diff
@@ -99,88 +99,86 @@ void SearchBar::searchTextChangedCallback(GtkEntry* entry, SearchBar* searchBar)
+    size_t count = control->getDocument()->getPageCount();
     if (count < 2) {
         // Nothing to do
         return;
     }
 
     MainWindow* win = control->getWindow();
-    int x = page + 1;
+    size_t searchedPage = currentPage + 1;
     GtkWidget* searchTextField = win->get("searchTextField");
     const char* text = gtk_entry_get_text(GTK_ENTRY(searchTextField));
     GtkWidget* lbSearchState = win->get("lbSearchState");
     if (*text == 0) {
         return;
     }
 
-    if (x >= count) {
-        x = 0;
+    if (searchedPage >= count) {
+        searchedPage = 0;
     }
 
-    double top = 0;
-    int occures = 0;
+    double yOfUpperMostMatch = 0;
+    size_t occurences = 0;
 
-    while (x != page) {
+    while (searchedPage != currentPage) {
 
-        bool found = control->searchTextOnPage(text, x, &occures, &top);
+        bool found = control->searchTextOnPage(text, searchedPage, &occurences, &yOfUpperMostMatch);
         if (found) {
-            control->getScrollHandler()->scrollToPage(x, top);
-            gtk_label_set_text(GTK_LABEL(lbSearchState),
-                               (occures == 1 ? FC(_F("Text found once on page {1}") % (x + 1)) :
-                                               FC(_F("Text found {1} times on page {2}") % occures % (x + 1))));
+            control->getScrollHandler()->scrollToPage(searchedPage, yOfUpperMostMatch);
+            gtk_label_set_text(
+                    GTK_LABEL(lbSearchState),
+                    (occurences == 1 ? FC(_F("Text found once on page {1}") % (searchedPage + 1)) :
+                                       FC(_F("Text found {1} times on page {2}") % occurences % (searchedPage + 1))));
             return;
         }
 
-        x++;
-        if (x >= count) {
-            x = 0;
+        searchedPage++;
+        if (searchedPage >= count) {
+            searchedPage = 0;
         }
     }
 
     gtk_label_set_text(GTK_LABEL(lbSearchState), _("Text not found, searched on all pages"));
 }
 
 void SearchBar::searchPrevious() {
-    int page = control->getCurrentPageNo();
-    int count = control->getDocument()->getPageCount();
+    size_t currentPage = control->getCurrentPageNo();
+    size_t count = control->getDocument()->getPageCount();
     if (count < 2) {
         // Nothing to do
         return;
     }
 
     MainWindow* win = control->getWindow();
-    int x = page - 1;
+    size_t searchedPage = currentPage == 0 ? count - 1 : currentPage - 1;
     GtkWidget* searchTextField = win->get("searchTextField");
     const char* text = gtk_entry_get_text(GTK_ENTRY(searchTextField));
     GtkWidget* lbSearchState = win->get("lbSearchState");
     if (*text == 0) {
         return;
     }
 
-    if (x < 0) {
-        x = count - 1;
-    }
-
-    double top = 0;
-    int occures = 0;
+    double yOfUpperMostMatch = 0;
+    size_t occurences = 0;
 
-    while (x != page) {
+    while (searchedPage != currentPage) {
```

Comment:
```suggestion
    // Search backwards through the pages, wrapping around if needed.
    while (searchedPage != currentPage) {
```

Reply:
Could also simplify into a for loop, but not sure if that would be more readable.

Reply:
I refactored SearchPage::searchNext and searchPrevious to remove the code duplication, and I documented the functions.
I chose not the change the behaviour,: the current page is not searched.
Those functions are called when pressing the next/previous button on the search bar. This then finds the next/previous page with matches and scrolls to said page.
I think the behaviour is a little weird (it should first navigate through the matches of the current page), but changing this is out of scope.